packages/xr/src/XRManagerExtended.ts

```diff
@@ -89,16 +91,16 @@ export class XRManagerExtended extends XRManager {
     }
   }
 
-  override getFeatures<T extends XRFeature>(type: TFeatureConstructor<T>, out?: T[]): T[] {
+  override getFeatures(out?: XRFeature[]): XRFeature[] {
+    const { _features: features } = this;
+    const length = features.length;
     if (out) {
-      out.length = 0;
+      out.length = length;
     } else {
-      out = [];
+      out = new Array<XRFeature>(length);
     }
-    const { _features: features } = this;
-    for (let i = 0, n = features.length; i < n; i--) {
-      const feature = features[i];
-      feature instanceof type && out.push(feature);
+    for (let i = 0; i < length; i++) {
+      out[i] = features[i];
```

Comment:
Generics are cancelled because only one of the same type can be added?

Reply:
Done.